src/rename.js

```diff
@@ -2,10 +2,12 @@ const assoc    = require('ramda/src/assoc')
 const converge = require('ramda/src/converge')
 const curry    = require('ramda/src/curry')
 const dissoc   = require('ramda/src/dissoc')
+const hasIn    = require('ramda/src/hasIn')
 const prop     = require('ramda/src/prop')
+const when     = require('ramda/src/when')
 
 // rename :: String -> String -> { k: v } -> { k: v }
-const rename = (from, to, obj) =>
-  converge(assoc(to), [ prop(from), dissoc(from) ])(obj)
+const rename = (frum, to, obj) =>
+  when(hasIn(frum), converge(assoc(to), [ prop(frum), dissoc(frum) ]))(obj)
```

Comment:
Do we need/want to have `obj` explicitly being passed in on the end?
```js
// rename :: String -> String -> { k: v } -> { k: v }
const rename = (frum, to) =>
  when(hasIn(frum), converge(assoc(to), [ prop(frum), dissoc(frum) ]))
```

Reply:
If we pulled in `crocks.curry`, I'd be ok with this.  What I don't want to match is Ramda's inconsistency in currying: some of their functions curry the final argument, and some (like `juxt` and `converge`) do not.  I would like to always curry the final argument like all the others, and to do so we can either use `R.curry` and pass the `obj` explicitly, or use `crocks.curry` and let it do the magic.

Reply:
Okay maybe some day. And crocks' `curry` isn't magic! It's gorgeous. I break down each line in a blog post here:
https://robertwpearce.com/blog/ramda-chops-function-currying.html